adb_graphics/datahandler/gribdata.py

```diff
@@ -1,254 +1,149 @@
-                  'Must be a list, range, or function call!')
+        return np.asarray(self.vspec.get("clevs", []))
 
     @staticmethod
     def date_to_str(date: datetime) -> str:
-
-        ''' Returns a formatted string (for graphic title) from a datetime
-        object'''
-
-        return date.strftime('%Y%m%d %H UTC')
-
-    @property
-    def field(self):
-
-        ''' Wrapper that calls get_field method for the current variable.
-        Returns the NioVariable object '''
-
-        return self._get_field(self.ncl_name(self.vspec))
-
-    def field_column_max(self, values, variable, level, **kwargs):
-
-        # pylint: disable=unused-argument
-
-        ''' Returns the column max of the values. '''
-
-        vals = self.values(name=variable, level=level, one_lev=False)
-        maxvals = vals.max(axis=0)
-
-        return maxvals
-
-    def field_sum(self, values, variable2, level2, **kwargs):
-
-        # pylint: disable=unused-argument
-
-        ''' Return the sum of the values. '''
-
-        value2 = self.values(name=variable2, level=level2)
-        sum2 = values + value2
-        value2.close()
-
-        return sum2
-
-    def field_diff(self, values, variable2, level2, **kwargs):
-
-        # pylint: disable=unused-argument
-
-        ''' Subtracts the values from variable2 from self.field. '''
-
-        value2 = self.values(name=variable2, level=level2)
-        diff = values - value2
-        value2.close()
-
-        return diff
-
-    def field_mean(self, values, variable, levels, global_levels, **kwargs):
-
-        # pylint: disable=unused-argument
-
-        ''' Returns the mean of the values. '''
-
-        fsum = np.zeros_like(values)
-
-        chosen_levels = global_levels if 'global' in self.model else levels
-        for level in global_levels:
-            val_lev = self.values(name=variable, level=level)
-            fsum = fsum + val_lev
-            val_lev.close()
-
-        return fsum / len(chosen_levels)
-
-    def _get_data_levels(self, vertical_dim):
-
-        ''' Return a list of vertical dimension values corresponding to the
-        requested vertical dimension to get the values of those dimensions '''
-
-        fcst_hr = 0 if self.ds.sizes.get('fcst_hr', 0) <= 1 else int(self.fhr)
-
-        ret = []
-        for dim in [var for var in self.ds.variables \
-                if vertical_dim in var]:
-
-            # Get the current forecast hour slice, if it's in the dataset
-            selector = {'fcst_hr': fcst_hr} if 'fcst_hr' in self.ds[dim].dims else {}
-            ret.append(self.ds[dim].sel(**selector).values)
-        return ret
-
-    def _get_field(self, ncl_name):
-
-        ''' Given an ncl_name, return the NioVariable object. '''
-
-        try:
-            field = self.ds[ncl_name.format(level_type=self.level_type)]
-        except KeyError:
-            raise errors.GribReadError(f'{ncl_name}')
-        return field
-
-    def _get_level(self, field, level, spec, **kwargs):
```

Comment:
This was a convoluted way to retrieve the level that may have been necessitated by pynio. It's removed with the cfgrib implementation.